### ISSUE TITLE
Methods on class inherited from Enum not showing

### ISSUE THREAD

**Issue by mbhall88** [pdoc3/pdoc#200]
## Expected Behavior
Defined methods on a class inherited from `enum.Enum` should be documented.

### Actual Behavior
Methods on a class inherited from `enum.Enum` are not documented.

### Steps to Reproduce

`tag.py`
```py
from enum import Enum
from typing import Set

class TagTypes(Enum):
    """An `Enum` used to collate all possible `Tag` types.

    The recommended way of initialising a `TagTypes` object is via the factory
    constructor `TagTypes.from_char`.
    """
    Character = 1
    Integer = 2
    RealNumber = 3
    String = 4

    @staticmethod
    def chars() -> Set[str]:
        """Returns a set of all the characters supported by this enum."""
        return {tag_type.value.char for tag_type in TagTypes}

    @staticmethod
    def from_char(char: str) -> "TagTypes":
        """Initialise a `TagTypes` from a character (`str`).
        """
        for tag_type in TagTypes:
            if char == tag_type.value.char:
                return tag_type
        raise ValueError(
            f"{char} is not in the set of known tag types {TagTypes.chars()}"
        )

```

I generate docs using `pdoc --html -f  <module>` and get the following warning also

```
/home/michael/.cache/pypoetry/virtualenvs/pafpy-Wmqy0Q2v-py3.6/lib/python3.6/site-packages/pdoc/html_helpers.py:498: ReferenceWarning: Code reference `TagTypes.from_char` in module "pafpy.tag" does not match any documented object.
  linked = re.sub(r'[a-zA-Z_]\w*(?:\.[a-zA-Z_]\w*)*(?:\(\))?', handle_refname, code_span)
```

I also tried `pdoc --html -f --config show_inherited_members=True <module>` but it didn't change anything.

When the docs open, I see the `TagTypes` class and its variants, but neither of the methods `chars()` or `from_char()`

### Additional info

- pdoc version: 0.8.1

**Comment by kernc** [comment]
`TagTypes(Enum)` is a type of class. @mbhall88 Welcome to investigate, why its static methods aren't picked as its public objects.
https://github.com/pdoc3/pdoc/blob/048919ece483e0609094cee2460d3467d2e0efc3/pdoc/__init__.py#L854-L881

**Comment by mbhall88** [comment]
Man, I got deep in this.  

**tl;dr** [This SO](https://stackoverflow.com/a/61490721/5299417) answer sums it up nicely.  

In `pdoc`, you run `inspect.getmembers()` here https://github.com/pdoc3/pdoc/blob/048919ece483e0609094cee2460d3467d2e0efc3/pdoc/__init__.py#L875

Using the running example above, when I do this I get

```py
>>> [_name for _name, _ in inspect.getmembers(TagTypes)]
['Character', 'Integer', 'RealNumber', 'String', '__class__', '__doc__', '__members__', '__module__', 'name', 'value']
```

So, we see that `chars()` and `from_char` aren't present... I also checked to see whether it was the fact that they were defined as static methods, but that isn't the problem

```py
from enum import Enum
from typing import Set

class TagTypes(Enum):
    """An `Enum` used to collate all possible `Tag` types.

    The recommended way of initialising a `TagTypes` object is via the factory
    constructor `TagTypes.from_char`.
    """
    Character = 1
    Integer = 2
    RealNumber = 3
    String = 4

    @staticmethod
    def chars() -> Set[str]:
        """Returns a set of all the characters supported by this enum."""
        return {tag_type.value.char for tag_type in TagTypes}

    @staticmethod
    def from_char(char: str) -> "TagTypes":
        """Initialise a `TagTypes` from a character (`str`).
        """
        for tag_type in TagTypes:
            if char == tag_type.value.char:
                return tag_type
        raise ValueError(
            f"{char} is not in the set of known tag types {TagTypes.chars()}"
        )

    def func(self):
        return self

>>> [_name for _name, _ in inspect.getmembers(TagTypes)]
['Character', 'Integer', 'RealNumber', 'String', '__class__', '__doc__', '__members__', '__module__', 'name', 'value']
```

I played around with defining `__getattr__`, `__dir__` and a bunch of other magic methods but to no avail. This seems to be well above my level of python. The only solution I found for getting these methods was by adapting some code from the above SO answer.

```py
def get_items(cls):
    lst = cls.__dict__.keys()
    try:
        res = [ (n, getattr(cls,n)) for n in lst ]
    except Exception as e:
        res = (Exception, type(e), e)
        pass
    return res
>>> get_items(TagTypes)
[('_generate_next_value_', <function Enum._generate_next_value_ at 0x7f622ba67d90>), ('__module__', '__main__'), ('__doc__', 'An `Enum` used to collate all possible `Tag` types.\n\n    The recommended way of initialising a `TagTypes` object is via the factory\n    constructor `TagTypes.from_char`.\n    '), ('chars', <function TagTypes.chars at 0x7f6224b47f28>), ('from_char', <function TagTypes.from_char at 0x7f6224b4c048>), ('func', <function TagTypes.func at 0x7f6224b4c0d0>), ('_member_names_', ['Character', 'Integer', 'RealNumber', 'String']), ('_member_map_', OrderedDict([('Character', <TagTypes.Character: 1>), ('Integer', <TagTypes.Integer: 2>), ('RealNumber', <TagTypes.RealNumber: 3>), ('String', <TagTypes.String: 4>)])), ('_member_type_', <class 'object'>), ('_value2member_map_', {1: <TagTypes.Character: 1>, 2: <TagTypes.Integer: 2>, 3: <TagTypes.RealNumber: 3>, 4: <TagTypes.String: 4>}), ('Character', <TagTypes.Character: 1>), ('Integer', <TagTypes.Integer: 2>), ('RealNumber', <TagTypes.RealNumber: 3>), ('String', <TagTypes.String: 4>), ('__new__', <function Enum.__new__ at 0x7f622ba67d08>)]
>>> # as opposed to the current way
>>> inspect.getmembers(TagTypes)
[('Character', <TagTypes.Character: 1>), ('Integer', <TagTypes.Integer: 2>), ('RealNumber', <TagTypes.RealNumber: 3>), ('String', <TagTypes.String: 4>), ('__class__', <class 'enum.EnumMeta'>), ('__doc__', 'An `Enum` used to collate all possible `Tag` types.\n\n    The recommended way of initialising a `TagTypes` object is via the factory\n    constructor `TagTypes.from_char`.\n    '), ('__members__', mappingproxy(OrderedDict([('Character', <TagTypes.Character: 1>), ('Integer', <TagTypes.Integer: 2>), ('RealNumber', <TagTypes.RealNumber: 3>), ('String', <TagTypes.String: 4>)]))), ('__module__', '__main__'), ('name', <types.DynamicClassAttribute object at 0x7f622ba62978>), ('value', <types.DynamicClassAttribute object at 0x7f622ba629e8>)]
```
Not sure how feasible this is, but that's about the best I can do with the time I have available at the moment.
